pkg/controllers/user/rbac/namespace_handler.go

```diff
@@ -118,6 +119,34 @@ func (n *nsLifecycle) ensurePRTBAddToNamespace(ns *v1.Namespace) (bool, error) {
 			return false, errors.Wrapf(err, "couldn't ensure binding %v in %v", prtb.Name, ns.Name)
 		}
 	}
+
+	if ns.Labels["field.cattle.io/projectId"] == "" {
+		return hasPRTBs, nil
+	}
+
+	prtbSet := map[string]string{}
+	prtbList := n.m.prtbIndexer.List()
```

Comment:
You're miss using indexer here.
you should define an indexer function like this:
```
func rtbByUID(obj interface{}) ([]string, error) {
	var idx string
	switch rtb := obj.(type) {
	case *v3.ProjectRoleTemplateBinding:
		return []string{string(rtb.UID)}, nil
	case *v3.ClusterRoleTemplateBinding:
		return []string{string(rtb.UID)}, nil
	}
	return nil, nil
}
```
register it in the indexer like this:
```
rtbByClusterAndRoleTemplateIndex: rtbByClusterAndRoleTemplateName,
+rtbByUIDIndex: rtbByUID,
```
and use it like this:
```
prtbs, err := c.m.prtbIndexer.ByIndex(rtbByClusterAndRoleTemplateIndex, prtbUIDFromRoleBinding)
if err != nil {
    return obj, err
}
```